From fce7500452f2aa0e27532f6c585bcdc7fe9b00e9 Mon Sep 17 00:00:00 2001
From: Liron Himi <lironh@marvell.com>
Date: Mon, 17 Dec 2018 18:00:16 +0200
Subject: [PATCH 11/30] patches: fine tuning params for l2fwd and l3fwd for a3k

Adding -a3k suffix to application executable name:
- l2fwd-a3k
- l3fwd-a3k

Change application setting to following:

#define MAX_PKT_BURST 20
#define RTE_TEST_RX_DESC_DEFAULT 512
#define RTE_TEST_TX_DESC_DEFAULT 512

Change-Id: I87c513c2c85bda326aa3e5f576154a3b0e226537
Signed-off-by: Liron Himi <lironh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61889
Reviewed-by: Alan Winkowski <walan@marvell.com>
(cherry picked from commit a7149e48ccd18284e3fa07db5fd26937d9c4d8c8)
Reviewed-on: http://vgitil04.il.marvell.com:8080/61961
Reviewed-on: https://sj1git1.cavium.com/1189
Reviewed-by: Alan Winkowski
Reviewed-by: Liron Himi
Tested-by: Liron Himi
---
 .../0001-examples-l2fwd-tune-params-for-a3k.patch  | 53 +++++++++++++++++++
 .../0001-examples-l3fwd-tune-params-for-a3k.patch  | 59 ++++++++++++++++++++++
 2 files changed, 112 insertions(+)
 create mode 100644 patches/0001-examples-l2fwd-tune-params-for-a3k.patch
 create mode 100644 patches/0001-examples-l3fwd-tune-params-for-a3k.patch

diff --git a/patches/0001-examples-l2fwd-tune-params-for-a3k.patch b/patches/0001-examples-l2fwd-tune-params-for-a3k.patch
new file mode 100644
index 0000000..59ba64e
--- /dev/null
+++ b/patches/0001-examples-l2fwd-tune-params-for-a3k.patch
@@ -0,0 +1,53 @@
+From 4e2fe11f3051e1bd204ecd466a3371236766093f Mon Sep 17 00:00:00 2001
+From: Natalie Samsonov <nsamsono@marvell.com>
+Date: Tue, 14 Aug 2018 22:10:23 +0300
+Subject: [PATCH 1/1] examples/l2fwd: tune params for a3k
+
+Creates executable with name: l2fwd-a3k
+
+Change-Id: I2d0991104da95cbb0daefa20edeaf75c36344438
+Signed-off-by: Natalie Samsonov <nsamsono@marvell.com>
+---
+ examples/l2fwd/Makefile | 2 +-
+ examples/l2fwd/main.c   | 6 +++---
+ 2 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/examples/l2fwd/Makefile b/examples/l2fwd/Makefile
+index a8a47ad..18db75f 100644
+--- a/examples/l2fwd/Makefile
++++ b/examples/l2fwd/Makefile
+@@ -2,7 +2,7 @@
+ # Copyright(c) 2010-2014 Intel Corporation
+ 
+ # binary name
+-APP = l2fwd
++APP = l2fwd-3k
+ 
+ # all source are stored in SRCS-y
+ SRCS-y := main.c
+diff --git a/examples/l2fwd/main.c b/examples/l2fwd/main.c
+index e4a4a7c..f0cf9b8 100644
+--- a/examples/l2fwd/main.c
++++ b/examples/l2fwd/main.c
+@@ -46,15 +46,15 @@ static int mac_updating = 1;
+ 
+ #define RTE_LOGTYPE_L2FWD RTE_LOGTYPE_USER1
+ 
+-#define MAX_PKT_BURST 32
++#define MAX_PKT_BURST 20
+ #define BURST_TX_DRAIN_US 100 /* TX drain every ~100us */
+ #define MEMPOOL_CACHE_SIZE 256
+ 
+ /*
+  * Configurable number of RX/TX ring descriptors
+  */
+-#define RTE_TEST_RX_DESC_DEFAULT 1024
+-#define RTE_TEST_TX_DESC_DEFAULT 1024
++#define RTE_TEST_RX_DESC_DEFAULT 512
++#define RTE_TEST_TX_DESC_DEFAULT 512
+ static uint16_t nb_rxd = RTE_TEST_RX_DESC_DEFAULT;
+ static uint16_t nb_txd = RTE_TEST_TX_DESC_DEFAULT;
+ 
+-- 
+2.7.4
+
diff --git a/patches/0001-examples-l3fwd-tune-params-for-a3k.patch b/patches/0001-examples-l3fwd-tune-params-for-a3k.patch
new file mode 100644
index 0000000..80330e1
--- /dev/null
+++ b/patches/0001-examples-l3fwd-tune-params-for-a3k.patch
@@ -0,0 +1,59 @@
+From 868e895711c355fab145ff4a7a1d63455be8be79 Mon Sep 17 00:00:00 2001
+From: Natalie Samsonov <nsamsono@marvell.com>
+Date: Tue, 14 Aug 2018 22:13:43 +0300
+Subject: [PATCH 1/1] examples/l3fwd: tune params for a3k
+
+Creates executable with name: l3fwd-a3k
+
+Change-Id: I452ba8ba992a68c5221976a45b665a8797e09a27
+Signed-off-by: Natalie Samsonov <nsamsono@marvell.com>
+---
+ examples/l3fwd/Makefile | 2 +-
+ examples/l3fwd/l3fwd.h  | 2 +-
+ examples/l3fwd/main.c   | 4 ++--
+ 3 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/examples/l3fwd/Makefile b/examples/l3fwd/Makefile
+index cccdd9d..474a393 100644
+--- a/examples/l3fwd/Makefile
++++ b/examples/l3fwd/Makefile
+@@ -2,7 +2,7 @@
+ # Copyright(c) 2010-2016 Intel Corporation
+ 
+ # binary name
+-APP = l3fwd
++APP = l3fwd-a3k
+ 
+ # all source are stored in SRCS-y
+ SRCS-y := main.c l3fwd_lpm.c l3fwd_em.c
+diff --git a/examples/l3fwd/l3fwd.h b/examples/l3fwd/l3fwd.h
+index c962dea..4e3e584 100644
+--- a/examples/l3fwd/l3fwd.h
++++ b/examples/l3fwd/l3fwd.h
+@@ -15,7 +15,7 @@
+ #define NO_HASH_MULTI_LOOKUP 1
+ #endif
+ 
+-#define MAX_PKT_BURST     32
++#define MAX_PKT_BURST     20
+ #define BURST_TX_DRAIN_US 100 /* TX drain every ~100us */
+ 
+ #define MAX_RX_QUEUE_PER_LCORE 16
+diff --git a/examples/l3fwd/main.c b/examples/l3fwd/main.c
+index e7111fa..ed46e03 100644
+--- a/examples/l3fwd/main.c
++++ b/examples/l3fwd/main.c
+@@ -50,8 +50,8 @@
+ /*
+  * Configurable number of RX/TX ring descriptors
+  */
+-#define RTE_TEST_RX_DESC_DEFAULT 1024
+-#define RTE_TEST_TX_DESC_DEFAULT 1024
++#define RTE_TEST_RX_DESC_DEFAULT 512
++#define RTE_TEST_TX_DESC_DEFAULT 512
+ 
+ #define MAX_TX_QUEUE_PER_PORT RTE_MAX_ETHPORTS
+ #define MAX_RX_QUEUE_PER_PORT 128
+-- 
+2.7.4
+
-- 
2.7.4

